solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -1290,20 +1240,6 @@ public ResponseParser getParser() {
     return parser;
   }
 
-  /**
-   * Note: This setter method is <b>not thread-safe</b>.
-   *
-   * @param parser Default Response Parser chosen to parse the response if the parser were not
-   *     specified as part of the request.
-   * @see org.apache.solr.client.solrj.SolrRequest#getResponseParser()
-   * @deprecated use {@link Http2SolrClient.Builder#withResponseParser(ResponseParser)} instead
-   */
-  @Deprecated
-  public void setParser(ResponseParser parser) {
-    this.parser = parser;
-    updateDefaultMimeTypeForParser();
```

Comment:
@epugh this breaks update of default mime types functionality. code needs to go via the setter when updating the parser so  `updateDefaultMimeTypeForParser` is always called otherwise that set becomes stale, I am working on a patch for SOLR-16737 and will add it back. fyi @tflobbe

Reply:
opened https://github.com/apache/solr/pull/1587 with a fix for this.